contracts/Depository.sol

```diff
@@ -208,10 +211,15 @@ contract Depository is IErrorsTokenomics {
             revert LowerThan(vesting, MIN_VESTING);
         }
 
-        // Check for the expiration time overflow
-        uint256 expiry = block.timestamp + vesting;
-        if (expiry > type(uint32).max) {
-            revert Overflow(expiry, type(uint32).max);
+        // Check the vesting maximum limit value
+        if (vesting > MAX_VESTING) {
+            revert Overflow(vesting, MAX_VESTING);
+        }
+
+        // Check for the maturity time overflow for the current timestamp
+        uint256 maturity = block.timestamp + vesting;
+        if (maturity > type(uint32).max) {
+            revert Overflow(maturity, type(uint32).max);
```

Comment:
Check for the max vesting and maturity limits.

Reply:
@DavidMinarsch this checks for the possible time overflow in a distant future, whereas `MAX_VESTING` limits the vesting period itself.

Reply:
Yep I understand.